akka-contrib/src/main/scala/akka/contrib/pattern/DistributedPubSubMediator.scala

```diff
@@ -548,6 +574,10 @@ class DistributedPubSubMediator(
       content = bucket.content + (key -> ValueHolder(v, valueOption))))
   }
 
+  def getCurrentTopics(): Map[String, ActorRef] = {
+    context.children.map(ref ⇒ (URLDecoder.decode(ref.path.name, "utf-8"), ref)).toMap
+  }
```

Comment:
My apologies.  Started a new job and have gotten completely overwhelmed by
the workload.  I will try to do now as I can.

Could you explain a bit more what this code represents:

if key.startsWith(topicPrefix)
topic = key.substring(topicPrefix.length + 1)
if !topic.contains('/') // exclude group topics

Maybe I'm a bit confused about how things are stored in the ergistyr and
how this will help me get the current topics.

Thanks!

On Wed, Jan 7, 2015 at 9:32 AM, Patrik Nordwall <notifications@github.com>
wrote:

> In
> akka-contrib/src/main/scala/akka/contrib/pattern/DistributedPubSubMediator.scala
> <https://github.com/akka/akka/pull/16324#discussion-diff-22588268>:
>
> > @@ -548,6 +574,10 @@ class DistributedPubSubMediator(
> >        content = bucket.content + (key -> ValueHolder(v, valueOption))))
> >    }
> >
> > +  def getCurrentTopics(): Map[String, ActorRef] = {
> > +    context.children.map(ref ⇒ (URLDecoder.decode(ref.path.name, "utf-8"), ref)).toMap
> > +  }
>
> I think I have misguided you. Sorry. The child actors are only the topics
> that have been registered at a specific node. We want all replicated
> topics. We must grab them from the registry. Something like:
>
> def getCurrentTopics(): Set[String] = {
> val topicPrefix = self.path.toStringWithoutAddress
> (for {
> (_, bucket) ← registry
> (key, value) ← bucket.content
> if key.startsWith(topicPrefix)
> topic = key.substring(topicPrefix.length + 1)
> if !topic.contains('/') // exclude group topics
> } yield URLDecoder.decode(topic, "utf-8")).toSet
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/akka/akka/pull/16324/files#r22588268>.
>